src/samplereader/TSSampleReader.cpp

```diff
@@ -69,6 +69,18 @@ AP4_Result CTSSampleReader::ReadSample()
 {
   if (ReadPacket())
   {
+    //! @todo: there is something wrong on pts calculation,
+    //! m_ptsOffs have a value in seconds and so the substraction "m_pts - m_ptsOffs" looks to be inconsistent,
+    //! To have pts in seconds on m_pts must be: pts = GetDts() / 90000,
+    //! but packet PTS seem to be different from m_ptsOffs pts value, as if it did not include the period start
+    //! so the substraction "m_pts - m_ptsOffs" it is not a clear thing.
+    //! There is also something weird on HLS discontinuities (multiple chapters/periods)
+    //! where after a discontinuity the packet pts is lower than the last segment of previous period/discontinuity
+    //! this cause a VP resync e.g:
+    //!   debug <general>: CVideoPlayer::CheckContinuity - resync backward :1, prev:587175999.000000, curr:577170666.000000, diff:-10005333.000000
+    //!   debug <general>: CVideoPlayer::CheckContinuity - update correction: -10026666.000000
+    //! not sure if this is correct or not (tested with pluto-tv)
+    //! This code is present also on the others sample readers, that need to be verified
     m_dts = (GetDts() == PTS_UNSET) ? STREAM_NOPTS_VALUE : (GetDts() * 100) / 9;
     m_pts = (GetPts() == PTS_UNSET) ? STREAM_NOPTS_VALUE : (GetPts() * 100) / 9;
```

Comment:
pts/dts operations to me is full wrong, here and more likely also on all other demuxers/samplereaders
that result in a wrong pts/dts fed to kodi

Reply:
Hmmm if the test was on pluto tv the segment lengths are 10 seconds, and the resync is 10 seconds (plus looks like difference between video/audio)... maybe old value is sent initially after discontinuity?

I still remember there is a bug (IMO) in VP when demuxread can't fill the internal 8s buffer fast enough on a period change, which causes VP to think it's out of sync by the same amount of time the entire stream has been playing. But that doesn't look like the case here. Could be related though? Something not being reinitialised to 0 in VP on period change and VP takes care of it this way usually?

Reply:
> Could be related though? Something not being reinitialised to 0 in VP on period change and VP takes care of it this way usually?

imo its different if you place a log in to "m_pts" value
on pluto you can see for example (random values)
packet pts 10000
packet pts 20000
packet pts 30000
---- discont - ads
packet pts 10000
packet pts 20000
---- discont - return to movie
packet pts ... a more higher value, now i dont remember exaclty if continue from last of 30000

idk if there is a way to intercept in right way these disc changes to prevent forward/backward of the VP that cause the resync
I have little info to understand if this is correct or if it needs proper fix